### ISSUE TITLE
The 0.2.2 cross fmt -- check won't return

### ISSUE THREAD

**Issue by yawqi** [cross-rs/cross#872]
### Checklist

- [X] I've looked through the [issues and pull requests](https://github.com/cross-rs/cross/issues?q=) for similar reports

### Describe your issue

We use cross in our release GitHub action, the newer 0.2.2 version of cross leads to the workflow gets cancelled, it seems the `cross fmt -- check` won't return I guess. When we downgrade to 0.2.1, the CI runs smoothly. Thanks for you help!
```
2022-06-26T23:27:59.5600709Z     Finished release [optimized] target(s) in 1m 41s
2022-06-26T23:27:59.5757107Z   Installing /home/runner/.cargo/bin/cross-util
2022-06-26T23:27:59.5757743Z    Replacing /home/runner/.cargo/bin/cross
2022-06-26T23:27:59.5827284Z    Installed package `cross v0.2.2` (executable `cross-util`)
2022-06-26T23:27:59.5828540Z     Replaced package `cross v0.2.1` with `cross v0.2.2` (executable `cross`)
2022-06-26T23:27:59.8220785Z info: downloading component 'rustfmt'
2022-06-26T23:27:59.9856139Z info: installing component 'rustfmt'
2022-06-26T23:28:00.5485512Z info: downloading component 'clippy'
2022-06-26T23:28:00.6008808Z info: installing component 'clippy'
2022-06-26T23:28:00.9917376Z cross fmt -- --check
2022-06-26T23:43:37.3842981Z make: *** [Makefile:57: .format] Error 130
2022-06-26T23:43:38.4643297Z ##[error]Process completed with exit code 143.
```

### What target(s) are you cross-compiling for?

x86_64-unknown-linux-musl

### Which operating system is the host (e.g computer cross is on) running?

- [ ] macOS
- [ ] Windows
- [X] Linux / BSD
- [ ] other OS (specify in description)

### What architecture is the host?

- [X] x86_64 / AMD64
- [ ] arm32
- [ ] arm64 (including Mac M1)

### What container engine is cross using?

- [ ] docker
- [ ] podman
- [X] other container engine (specify in description)

### cross version

cross 0.2.2

### Example

https://github.com/dragonflyoss/image-service/runs/7056755026?check_suite_focus=true
https://github.com/dragonflyoss/image-service/runs/7063799025?check_suite_focus=true

**Comment by Alexhuszagh** [comment]
Did you set `CARGO=cross` by any chance? This was fixed in #868.

**Comment by yawqi** [comment]
> Did you set `CARGO=cross` by any chance? This was fixed in #868. `cross` falls back to `rustfmt` on the host since it doesn't make sense to run `rustfmt` in the image, so the only reason why it wouldn't return is if it was accidentally set into a loop, which we fixed in the above PR.

Yes, `make -e ARCH=$arch -e CARGO=cross static-release`, and use cross 0.2.1 works fine : ).

**Comment by Alexhuszagh** [comment]
> > Did you set `CARGO=cross` by any chance? This was fixed in #868. `cross` falls back to `rustfmt` on the host since it doesn't make sense to run `rustfmt` in the image, so the only reason why it wouldn't return is if it was accidentally set into a loop, which we fixed in the above PR.
> 
> Yes, `make -e ARCH=$arch -e CARGO=cross static-release`, and use cross 0.2.1 works fine : ).

See #868, and either install `cross` from git or unset `CARGO=cross`. We patched this bug earlier today, which was initially reported in #865.

**Comment by yawqi** [comment]
> > > Did you set `CARGO=cross` by any chance? This was fixed in #868. `cross` falls back to `rustfmt` on the host since it doesn't make sense to run `rustfmt` in the image, so the only reason why it wouldn't return is if it was accidentally set into a loop, which we fixed in the above PR.
> > 
> > 
> > Yes, `make -e ARCH=$arch -e CARGO=cross static-release`, and use cross 0.2.1 works fine : ).
> 
> See #868, and either install `cross` from git or unset `CARGO=cross`. We patched this bug earlier today, which was initially reported in #865.
> 
> We changed our logic to process `CARGO` and prioritize it over `cargo` when running commands on the host. Unfortunately, since many people use `CARGO=cross`, this meant that many people had infinite loops. Since `CARGO` is only set by `cargo`, but not used by it during configuration, we've reverted it back to ignore `CARGO` like before.

Thanks a lot! 👍
